src/client/content_cao.cpp

```diff
@@ -520,6 +523,22 @@ void GenericCAO::removeFromScene(bool permanent)
 	}
 }
 
+void GenericCAO::setSceneNodeMaterial(scene::ISceneNode *node,
```

Comment:
Why is the scene node pointer passed as an argument? Wouldn't `GenericCAO::getSceneNode` work?

https://github.com/minetest/minetest/blob/4291961e09db7851fbc940eac58a1b919668d935/src/client/content_cao.cpp#L420-L438